common/lib/protocol-definitions/src/clients.ts

```diff
@@ -3,66 +3,181 @@
  * Licensed under the MIT License.
  */
 
+import { ScopeType } from "./scopes";
 import { IUser } from "./users";
 
+/**
+ * A client's connection mode - either view-only ("read") or allowing edits ("write").
+ *
+ * @remarks
+ *
+ * Note: a user's connection mode is dependent on their permissions.
+ * E.g. a user with read-only permissions will not be allowed a "write" connection mode.
+ */
 export type ConnectionMode = "write" | "read";
 
+/**
+ * Capabilities of a Client.
+ * In particular, whether or not the client is {@link ICapabilities.interactive}.
+ */
 export interface ICapabilities {
+	/**
+	 * Indicates if the client represents a potentially interactive session with a user (if 'true') or if it's
+	 * a "system entity" (if 'false').
+	 *
+	 * @remarks
+	 *
+	 * The only "system entity" scenario at the moment is the
+	 * {@link https://fluidframework.com/docs/concepts/summarizer/ | summarizer client}.
+	 */
 	interactive: boolean;
 }
+
+/**
+ * The kind of client connected to the service.
+ */
+export type ClientType = "client" | "summarizer";
+
+/**
+ * {@link IClient} connection / environment metadata.
+ */
 export interface IClientDetails {
+	/**
+	 * {@inheritDoc ICapabilities}
+	 */
 	capabilities: ICapabilities;
-	type?: string;
+
+	/**
+	 * The kind of client being described.
+	 *
+	 * `undefined` indicates that the kind could not be determined.
+	 *
+	 * @remarks
+	 *
+	 * General `string` values are allowed for type-wise backwards compatibility, but this support
```

Comment:
Is this _intended_ to always be one of these 2 values? They appear to be the only ones used between fluid-framework and bohemia / this repo makes assumptions based on the 2 values, but it isn't immediately clear if that is architecturally intended.